src/System.Text.Json/tests/JsonNodeTests.JsonNumber.cs

```diff
@@ -0,0 +1,632 @@
+            Assert.Equal(value, jsonNumber.GetUInt16());
+            Assert.True(jsonNumber.TryGetUInt16(out result));
+            Assert.Equal(value, result);
+
+            // String constructor:
+            jsonNumber = new JsonNumber(value.ToString());
+            Assert.Equal(value, jsonNumber.GetUInt16());
+            Assert.True(jsonNumber.TryGetUInt16(out result));
+            Assert.Equal(value, result);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(17)]
+        [InlineData(uint.MaxValue)]
+        public static void TestUInt32(uint value)
+        {
+            // Default constructor:
+            var jsonNumber = new JsonNumber();
+            jsonNumber.SetUInt32(value);
+            Assert.Equal(value, jsonNumber.GetUInt32());
+            Assert.True(jsonNumber.TryGetUInt32(out uint result));
+            Assert.Equal(value, result);
+
+            // Numeric type constructor:
+            jsonNumber = new JsonNumber(value);
+            Assert.Equal(value, jsonNumber.GetUInt32());
+            Assert.True(jsonNumber.TryGetUInt32(out result));
+            Assert.Equal(value, result);
+
+            // Implicit cast:
+            jsonNumber = value;
+            Assert.Equal(value, jsonNumber.GetUInt32());
+            Assert.True(jsonNumber.TryGetUInt32(out result));
+            Assert.Equal(value, result);
+
+            // String constructor:
+            jsonNumber = new JsonNumber(value.ToString());
+            Assert.Equal(value, jsonNumber.GetUInt32());
+            Assert.True(jsonNumber.TryGetUInt32(out result));
+            Assert.Equal(value, result);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(17)]
+        [InlineData(ulong.MaxValue)]
+        public static void TestUInt64(ulong value)
+        {
+            // Default constructor:
+            var jsonNumber = new JsonNumber();
+            jsonNumber.SetUInt64(value);
+            Assert.Equal(value, jsonNumber.GetUInt64());
+            Assert.True(jsonNumber.TryGetUInt64(out ulong result));
+            Assert.Equal(value, result);
+
+            // Numeric type constructor:
+            jsonNumber = new JsonNumber(value);
+            Assert.Equal(value, jsonNumber.GetUInt64());
+            Assert.True(jsonNumber.TryGetUInt64(out result));
+            Assert.Equal(value, result);
+
+            // Implicit cast:
+            jsonNumber = value;
+            Assert.Equal(value, jsonNumber.GetUInt64());
+            Assert.True(jsonNumber.TryGetUInt64(out result));
+            Assert.Equal(value, result);
+
+            // String constructor:
+            jsonNumber = new JsonNumber(value.ToString());
+            Assert.Equal(value, jsonNumber.GetUInt64());
+            Assert.True(jsonNumber.TryGetUInt64(out result));
+            Assert.Equal(value, result);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        public static void TestDecimal(decimal value)
```

Comment:
The theory on this one would probably need to take a `(string valueString)`, then immediately do `decimal value = decimal.Parse(valueString)`.   Alternatively use MemberData and make an array of literals using the `m` suffix.  e.g. (`3.141592658m`)